src/figuresClasses.ts

```diff
@@ -1,19 +1,82 @@
 export interface Figure {
-
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
-export class Triangle {
+enum Shapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+function checkSides(...sides: number[]): void {
+  if (sides.some((side) => side <= 0)) {
+    throw new Error('Value should be greater than 0 !');
+  }
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shapes.Triangle;
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    checkSides(a, b, c);
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error("The sides don't form a valid triangle");
+    }
+  }
+
+  getArea(): number {
+    const semiperimeter = (this.a + this.b + this.c) / 2;
 
+    return Math.floor(
+      Math.sqrt(
+        semiperimeter
+        * (semiperimeter - this.a)
+        * (semiperimeter - this.b)
+        * (semiperimeter - this.c),
+      ) * 100,
+    ) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape = Shapes.Circle;
 
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    checkSides(radius);
+  }
+
+  getArea(): number {
+    return Math.floor(Math.PI * this.radius ** 2 * 100) / 100;
```

Comment:
Would be nice to move rounding to separate helper function, you've used this logic in all classes